### PR TITLE
Added test cases as per #1161

### LINKED ISSUE #1161: bowling: update tests and add version file
The bowling tests should be updated to exactly match the canonical data. Also a version file should be added to match the canonical data version.

### PR DESCRIPTION
**Pull request by TimoleonLatinopoulos**
Tried to fix #1161. [Canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/bowling/canonical-data.json)

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by TimoleonLatinopoulos** [comment]
Ok let me have another go on this. @sjwarner-bp

**Comment by sjwarner-bp** [comment]
Sure thing! If you continue to push changes, this PR should automatically update :slightly_smiling_face:

**Comment by TimoleonLatinopoulos** [comment]
I didn't know that. I'll make sure to run the tests on my computer first before uploading anything.

**Comment by sjwarner-bp** [comment]
I found it super useful when starting out here :slightly_smiling_face:

File: exercises/bowling/.meta/src/reference/java/BowlingGame.java [hunk 1]
```diff
@@ -23,12 +23,12 @@ int score() {
                 if (rolls.size() <= frameIndex + 2) {
                     throw new IllegalStateException("Score cannot be taken until the end of the game");
                 }
-
+
```
Comment (sjwarner-bp) [inline exercises/bowling/.meta/src/reference/java/BowlingGame.java]: This looks perfect, if you could just look at removing the extra space on the indicated lines I'll merge this in 🙂

File: exercises/bowling/.meta/src/reference/java/BowlingGame.java [hunk 2]
```diff
@@ -23,12 +23,12 @@ int score() {
...
                 }
-
+                
                 int strikeBonus = strikeBonus(frameIndex);
-                if (strikeBonus > MAXIMUM_FRAME_SCORE && !isStrike(frameIndex + 1)) {
+                if (strikeBonus > MAXIMUM_FRAME_SCORE && !isStrike(frameIndex + 1) || spareBonus(frameIndex) > 10) {
                     throw new IllegalStateException("Pin count exceeds pins on the lane");
-                }
-
+                }
```
Comment (sjwarner-bp) [inline exercises/bowling/.meta/src/reference/java/BowlingGame.java]: Here

File: exercises/bowling/.meta/src/reference/java/BowlingGame.java [hunk 3]
```diff
@@ -23,12 +23,12 @@ int score() {
...
-
+                
                 int strikeBonus = strikeBonus(frameIndex);
-                if (strikeBonus > MAXIMUM_FRAME_SCORE && !isStrike(frameIndex + 1)) {
+                if (strikeBonus > MAXIMUM_FRAME_SCORE && !isStrike(frameIndex + 1) || spareBonus(frameIndex) > 10) {
                     throw new IllegalStateException("Pin count exceeds pins on the lane");
-                }
-
+                }       
+
```
Comment (sjwarner-bp) [inline exercises/bowling/.meta/src/reference/java/BowlingGame.java]: Here too 🙂

**Comment by TimoleonLatinopoulos** [comment]
The reverted commits are not relevant to this PR. I was trying to open a new one for another issue, but it seems that this one needs to close first. :)

**Comment by TimoleonLatinopoulos** [comment]
Should I close this pull request or should I wait for merging @sjwarner-bp ?

**Comment by sjwarner** [comment]
@TimoleonLatinopoulos sorry, I got distracted in the middle of this!

I'll merge it now 🙂 

You can have more than one PR open at any one time, and using the branches linked above this should separate your commits out nicely 🙂

**Comment by sjwarner-bp** [comment]
@TimoleonLatinopoulos sorry, I got distracted in the middle of this!

I'll merge it now 🙂 

You can have more than one PR open at any one time, and using the branches linked above this should separate your commits out nicely 🙂

**Pull request merged**